controlplane/internal/controller/clusterdeployment_controller_test.go

```diff
@@ -188,4 +188,186 @@ var _ = Describe("ClusterDeployment Controller", func() {
 		k8sClient = nil
 		controllerReconciler = nil
 	})
+
+	Context("setACICapabilities", func() {
+		When("It's a SNO cluster and there's no capabilities defined in the OACP spec", func() {
+			It("should not set capabilities with install config overrides annotation", func() {
+				oacp := &v1alpha2.OpenshiftAssistedControlPlane{
+					Spec: v1alpha2.OpenshiftAssistedControlPlaneSpec{
+						Replicas: int32(1),
+					},
+				}
+				aci := &hiveext.AgentClusterInstall{}
+				err := setACICapabilities(oacp, aci)
+				Expect(err).To(BeNil())
+				Expect(aci.Annotations).To(BeEmpty())
+			})
+		})
+		When("It's a SNO cluster and there's capabilities defined in the OACP spec", func() {
+			It("should have the exact listed capabilities set in the install config overrides annotation", func() {
+				oacp := &v1alpha2.OpenshiftAssistedControlPlane{
+					Spec: v1alpha2.OpenshiftAssistedControlPlaneSpec{
+						Replicas: int32(1),
+						Config: v1alpha2.OpenshiftAssistedControlPlaneConfigSpec{
+							AdditionalEnabledCapabilities: []string{"baremetal", "MachineAPI", "NodeTuning"},
+						},
+					},
+				}
+				aci := &hiveext.AgentClusterInstall{}
+				expectedAnnotation := `{"capabilities": {"baselineCapabilitySet": "None", "additionalEnabledCapabilities": ["baremetal","MachineAPI","NodeTuning"]}}"`
+
+				err := setACICapabilities(oacp, aci)
+				Expect(err).To(BeNil())
+				Expect(aci.Annotations).NotTo(BeEmpty())
+				Expect(aci.Annotations).To(HaveKey(InstallConfigOverrides))
+				Expect(aci.Annotations[InstallConfigOverrides]).To(Equal(expectedAnnotation))
+			})
+		})
+		When("It's a non-baremetal multi-node cluster and there's no capabilities defined in the OACP spec", func() {
+			It("should not set capabilities with install config overrides annotation", func() {
+				oacp := &v1alpha2.OpenshiftAssistedControlPlane{
+					Spec: v1alpha2.OpenshiftAssistedControlPlaneSpec{
+						Replicas: int32(3),
+					},
+				}
+				aci := &hiveext.AgentClusterInstall{}
+				err := setACICapabilities(oacp, aci)
+				Expect(err).To(BeNil())
+				Expect(aci.Annotations).To(BeEmpty())
+			})
+		})
+		When("It's a non-baremetal multi-node cluster and there's capabilities defined in the OACP spec", func() {
+			It("should have the exact listed capabilities set in the install config overrides annotation", func() {
+				oacp := &v1alpha2.OpenshiftAssistedControlPlane{
+					Spec: v1alpha2.OpenshiftAssistedControlPlaneSpec{
+						Replicas: int32(3),
+						Config: v1alpha2.OpenshiftAssistedControlPlaneConfigSpec{
+							AdditionalEnabledCapabilities: []string{"baremetal", "MachineAPI", "NodeTuning"},
+						},
+					},
+				}
+				aci := &hiveext.AgentClusterInstall{}
+				expectedAnnotation := `{"capabilities": {"baselineCapabilitySet": "None", "additionalEnabledCapabilities": ["baremetal","MachineAPI","NodeTuning"]}}"`
+
+				err := setACICapabilities(oacp, aci)
```

Comment:
In general, it'd be best not testing private methods. We should provide input and test the expected output (i.e. oacp specs, and expect the ACI to have this annotation) without directly calling this private method

Reply:
Ah right, sorry I forgot about not testing private methods. Will update. Thanks!

Reply:
Will fully update later, but keeping these for now to verify the function does what it's supposed to.